api/objects.proto

```diff
@@ -222,3 +222,54 @@ message Cluster {
 	// a new key is allocated on key rotation.
 	uint64 encryption_key_lamport_clock = 6;
 }
+
+// Secret represents a conceptual secret that should be passed to a container.
+// It can contain many versions of secret data, since conceptually secret
+// credentials can be rotated.
+message Secret {
+	string id = 1 [(gogoproto.customname) = "ID"];
+
+	Meta meta = 2 [(gogoproto.nullable) = false];
+
+	// A secret has multiple versions, and each version is represented by a
+	// `SecretData` object and referenced by the `SecretData`'s ID.  This map
+	// stores `SecretData` objects keyed by ID, which lets us reference any
+	// particular version of a secret easily.
+	map<string, SecretData> secret_data = 3;
```

Comment:
We have timestamps to order them, but the objective here was to have the ability to very quickly retrieve the data for a specific version of a secret without iterating through all the `SecretDatas`